### PR TITLE
Split tck to base and rest-client

### PR DESCRIPTION
**Pull request by pavolloffay**
Split TCK to allow run it for implementors who do not support rest-client.

Now there are two tck modules
*  base - `microprofile-opentracing-tck`. Should we rename it to `microprofile-opentracing-tck-base`?
* rest-client `microprofile-opentracing-tck-rest-client`

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

**Comment by pavolloffay** [comment]
@fmhwong could you please review?

**Comment by pilhuhn** [comment]
Sounds good to me as well

File: spec/src/main/asciidoc/microprofile-opentracing.asciidoc [hunk 1]
```diff
@@ -168,10 +168,13 @@ Tracing in `javax.ws.rs.client.Client` has to be explicitly enabled by invoking
 The implementation might enable client tracing globally, in this case explicit configuration has no effect.
 
 ===== MicroProfile Rest Client
-Tracing for MicroProfile Rest Client is by default globally enabled and it can be disabled by
+Tracing for this client is by default globally enabled and it can be disabled by
 specifying `@Traced(false)` on the client interface or method. When it is specified on the client's
 interface tracing is disabled for all methods.
 
+Note that integration with MicroProfile Rest Client is not mandatory for vendors not implementing
```
Comment (pavolloffay) [inline spec/src/main/asciidoc/microprofile-opentracing.asciidoc]: @fmhwong @pilhuhn could you please have a look? I am adding this to clarify a parial support if the client spec is not supported by the vendor.\

Comment (pavolloffay) [reply spec/src/main/asciidoc/microprofile-opentracing.asciidoc]: We do not mention any version numbers in readme, I think it's ok to go without the version for the consistency. However I will add the version to the changelog at least.

**Pull request merged**